rules/naming/src/Rector/Assign/RenameVariableToMatchGetMethodNameRector.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Naming\Rector\Assign;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ArrowFunction;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Expr\Variable;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\Naming\Naming\ExpectedNameResolver;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see \Rector\Naming\Tests\Rector\Assign\RenameVariableToMatchGetMethodNameRector\RenameVariableToMatchGetMethodNameRectorTest
+ */
+final class RenameVariableToMatchGetMethodNameRector extends AbstractRector
+{
+    /**
+     * @var ExpectedNameResolver
+     */
+    private $expectedNameResolver;
+
+    public function __construct(ExpectedNameResolver $expectedNameResolver)
+    {
+        $this->expectedNameResolver = $expectedNameResolver;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Rename variable to match get method name', [
+            new CodeSample(
+                <<<'PHP'
+class SomeClass {
+    public function run()
+    {
+        $a = $this->getRunner();
+    }
+}
+PHP
+,
+                <<<'PHP'
+class SomeClass {
+    public function run()
+    {
+        $runner = $this->getRunner();
+    }
+}
+PHP
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Assign::class];
+    }
+
+    /**
+     * @param Assign $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->expr instanceof ArrowFunction) {
+            return null;
+        }
+
+        /** @var MethodCall|StaticCall|FuncCall $callNode */
+        $callNode = $this->betterNodeFinder->findFirst($node->expr, function ($node) {
+            return $node instanceof MethodCall || $node instanceof StaticCall || $node instanceof FuncCall;
+        });
+
+        if ($callNode === null) {
+            return null;
+        }
+
+        $newName = $this->expectedNameResolver->resolveForMethodCall($callNode);
+        if ($newName === null || $this->isName($node, $newName)) {
+            return null;
+        }
+
+        /** @var Node\FunctionLike $classLikeNode */
+        $functionLikeNode = $this->betterNodeFinder->findFirstParentInstanceOf($node, Node\FunctionLike::class);
+        if ($functionLikeNode !== null) {
+            $nodes = $functionLikeNode->getStmts();
+        } else {
+            $functionLikeNode = $node->getAttribute(AttributeKey::PARENT_NODE);
+            $nodes = [];
+            /** @var Node\Stmt\Expression $expression */
+            while ($expression = $functionLikeNode->getAttribute(AttributeKey::NEXT_NODE)) {
+                $nodes[] = $expression;
+            }
```

Comment:
What exactly is doing this? Maybe extra a method with clear name

Reply:
This is for plain php script. In plain php script there is no parent node (or at least I didn't find how to get it) that is why here I'm getting all Expr this way after the node where there is an assignment.

I'll think about how to make it more understandable after two glasses of wine ;)

Reply:
> This is for plain php script.

Btw, no need to cover these cases. They're pretty edge ones and most of rules don't handle spaghetti code. 

There are rules that help to migrate spaghetti code to classes and PSR-4, then the best practise is to apply rules like these. So you can remove this case + test fixture.